x-pack/plugins/security_solution/cypress/objects/rule.ts

```diff
@@ -46,20 +52,21 @@ export interface CustomRule {
   description: string;
   dataSource: RuleDataSource;
   interval?: string;
-  severity: string;
-  riskScore: string;
-  tags: string[];
+  severity?: string;
+  riskScore?: string;
+  tags?: string[];
   timelineTemplate?: string;
-  referenceUrls: string[];
-  falsePositivesExamples: string[];
-  mitre: Mitre[];
-  note: string;
-  runsEvery: Interval;
-  lookBack: Interval;
-  timeline: CompleteTimeline;
-  maxSignals: number;
+  referenceUrls?: string[];
+  falsePositivesExamples?: string[];
+  mitre?: Mitre[];
+  note?: string;
+  runsEvery?: Interval;
+  lookBack?: Interval;
+  timeline?: CompleteTimeline;
+  maxSignals?: number;
   buildingBlockType?: string;
   exceptionLists?: Array<{ id: string; list_id: string; type: string; namespace_type: string }>;
+  actions?: Actions;
```

Comment:
@banderror I completely agree with you on this, not only for rules but for other types we have. That is something I thought as well but I didn't have the chance to jump and refactor the whole suite.

When the first test was implemented, the idea behind was for the type to represent more which fields are mandatory or not on the UI. As you could see, as more tests have been added this idea has disappeared having a mix that is not easy to maintain over time, so if reusing the types is going to make our lives easier. You have my upvote! :)

Reply:
++ Reusing common types make total sense to me. In case API types don't match the UI, we could try deriving the correct types from the schema using `Pick` or other utilities.